api/imapsync/update

```diff
@@ -43,28 +45,73 @@ if ($cmd eq 'update') {
     }
 
     umask 077;
-    open(FH, '>', "/var/lib/nethserver/secrets/imapsync-$input->{'name'}");
-    print FH $input->{'password'};
+    my ($login,$pass,$uid,$gid) = getpwnam('vmail');
+
+    # copy vmail password in readable place for vmail
+    my $source = '/var/lib/nethserver/secrets/vmail';
+    my $destination = '/var/lib/nethserver/imapsync/secrets/vmail';
+    copy ($source,$destination);
+
+    # copy remote password in readable place for vmail
+    open(SH, '>', "/var/lib/nethserver/imapsync/secrets/$input->{'name'}");
+    print SH $input->{'password'};
+    close SH;
+    chown $uid, $gid, glob("/var/lib/nethserver/imapsync/secrets/*");
+
+    # Find the correct security policy
+    my $crypt;
+    if ($input->{Security} eq 'tls') {
+        $crypt = '--tls1';
+    } elsif ($input->{Security} eq 'ssl') {
+        $crypt = '--ssl1';
+    } else {
+        $crypt = '';
+    }
+    # Find if I delete folder in the destination
+    my $delete2 = ($DeleteDestination eq 'enabled') ? '--delete2 --delete2folders':'';
+
+    # write the Env file for systemd
+    my $vars = qq(
+    LOCALUSER=$input->{'name'}
+    REMOTEHOSTNAME=$input->{'hostname'}
+    REMOTEPORT=$input->{'Port'}
+    SECURITY=$crypt
+    DELETE=$delete2
+    REMOTEUSERNAME=$input->{'username'});
+
+    open(FH, '>', "/var/lib/nethserver/imapsync/$input->{'name'}");
+    print FH $vars;
     close FH;
+    chown $uid, $gid, glob("/var/lib/nethserver/imapsync/*");
+
+    # start the service
+    system ("/usr/bin/systemctl  restart  imapsync\@$input->{'name'}");
```

Comment:
Remember `system LIST` to avoid unnecessary shell expansion!

```suggestion
    system (qw(/usr/bin/systemctl  restart),  "imapsync\@$input->{'name'}");
```

Reply:
done